src/ring/middleware/session_timeout.clj

```diff
@@ -23,17 +44,32 @@
          (if (map? timeout-response)
            (nil? timeout-handler)
            (ifn? timeout-handler))]}
-  (fn [request]
-    (let [session  (:session request {})
-          end-time (::idle-timeout session)]
-      (if (and end-time (< end-time (current-time)))
-        (assoc (or timeout-response (timeout-handler request)) :session nil)
-        (when-let [response (handler request)]
-          (let [session (:session response session)]
-            (if (nil? session)
-              response
-              (let [end-time (+ (current-time) timeout)]
-                (assoc response :session (assoc session ::idle-timeout end-time))))))))))
+  (fn
+    ([request]
+     (if (session-idle-expired? request)
```

Comment:
If we instead:

```clojure
(let [session (:session request {})]
  (if (session-idle-expired? session)
    ...))
```

Then we can reuse the request session when passing to `idle-session-timeout-response`. See my [prior example](https://github.com/ring-clojure/ring-session-timeout/pull/7#issuecomment-1463087485).